testgrid/cmd/updater/main.go

```diff
@@ -659,7 +473,7 @@ func ReadBuild(build Build) (*Column, error) {
 	go func() {
 		defer wg.Done()
 		defer close(artifacts) // No more artifacts
-		if err := listArtifacts(ctx, build, artifacts); err != nil {
+		if err := build.Artifacts(artifacts); err != nil {
```

Comment:
I know this looks scary but it is essentially:

* Move `readFinished()` `readStarted()` `listArtifacts()` and `listSuites()` into its own file
* Make it `build.Started()` instead of `readStarted(build)`, etc


So theoretically is simple, and we know from our simple refactoring of prow's k8s client that this is foolproof! On the other hand, no one is using this code in prod right now.

Reply:
The shade is so real ...

Reply:
I just want my PR reviewed!